lib/slack/bot.ex

```diff
@@ -87,11 +105,11 @@ defmodule Slack.Bot do
       token: token,
       me: rtm.self,
       team: rtm.team,
-      bots: rtm_list_to_map(rtm.bots),
-      channels: rtm_list_to_map(rtm.channels),
-      groups: rtm_list_to_map(rtm.groups),
-      users: rtm_list_to_map(rtm.users),
-      ims: rtm_list_to_map(rtm.ims)
+      bots: bot_to_map(Bots.info(%{token: token}) |> Map.get("bot")),
```

Comment:
This seems incorrect - `bot_to_map/1` wraps the response as a map with a key using`bot.id` - so there won't be any value under `"bot"`.

Reply:
it's meant to map to the original functionality as you can see from the test

Reply:
I see the intention, but it's broken.

Previously, `rtm.bots` seem to a list of map having `"id"` key, so using `rtm_list_to_map/1`, it returns a map with key of bot ids.

Now, `bot_to_map/1` does that - so applying `Map.get("bot")` would result in `nil`

Reply:
@chulkilee the [response](https://api.slack.com/methods/bots.info) returns a map within a map - so in order to get the `bot` property I need to go 1 layer deeper.

Reply:
Then shouldn't you do this?

```elixir
Bots.info(%{token: token}) |> Map.get("bot")
```

Please note that `bot_to_map/1` *wraps* the map.

```elixir
  defp bot_to_map(bot) do
    %{"#{bot.id}" => bot}
  end
```

Reply:
I'm not really understanding your question - that is what I'm doing. What are you suggesting I do differently?

Reply:
What the current code is doing is:

```
Bots.info(%{token: token} |> bot_to_map() |> Map.get("bot"))
```

not

```
Bots.info(%{token: token} |> Map.get("bot"))
```

Reply:
The code you presented causes the tests to fail. I don't think these refactoring changes are worth it if they aren't going to keep the code working properly.

Reply:
Mine was "suggestion", not working code. I hope you get the idea.